app/lib/dartdoc/models.dart

```diff
@@ -3,16 +3,112 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:convert';
 
 import 'package:json_annotation/json_annotation.dart';
 import 'package:meta/meta.dart';
 
+import '../shared/datastore.dart' as db;
 import '../shared/utils.dart' show jsonUtf8Encoder, utf8JsonDecoder;
 import '../shared/versions.dart' as versions;
 import 'storage_path.dart' as storage_path;
 
 part 'models.g.dart';
 
+/// Status values for [DartdocRecord].
+abstract class DartdocRecordStatus {
+  static const uploading = 'uploading';
+  static const ready = 'ready';
+  static const deleting = 'deleting';
+}
+
+@db.Kind(name: 'DartdocRecord', idType: db.IdType.String)
+class DartdocRecord extends db.ExpandoModel<String> {
+  String get uuid => id;
```

Comment:
unique identifier that identifies a specific execution of dartdoc.. maybe it could be called `runId`? and have a documentation string that says it's a uuid for a specific run of dartdoc.

Reply:
I like `runId`, not sure why we haven't used it before...